File: backend/DTOs/Project.cs
namespace Backend.DTOs;

public class CreateProjectDto
{
    public required string Name { get; set; }
    public string? Description { get; set; }
}

public class ProjectResponseDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    
    public Guid? CreatedBy { get; set; }
    public Guid? UpdatedBy { get; set; }
    
    public string CreatedByDisplayName { get; set; } = "System";
    public string UpdatedByDisplayName { get; set; } = "Never";
    
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: frontend-angular/src/app/core/models/project.ts
export interface Project {
  id: string;
  name: string;
  description?: string;
  workspaceId: string;
  workspaceName: string;
  createdBy?: string;
  updatedBy?: string;
  createdByDisplayName?: string;
  updatedByDisplayName?: string;
  createdAt: string;
  updatedAt?: string;
  isDeleted: boolean;
  deletedAt?: string;
  isPurgeable: boolean;
}

/** A move drops assignees the target workspace does not contain, so it reports how many. */
export interface MoveProjectResult {
  project: Project;
  unassignedTaskCount: number;
}
